# import libraries

In [2]:
import pandas as pd 

# get data

In [3]:
csv_path = 'C:\\Users\\jlfen\\OneDrive\\Documentos\\JoaoKasten\\005_applied_projects\\project.censo-dataviz\\src\\data\\census_data.csv'
df = pd.read_csv(csv_path)

## 1. Análise de Crescimento Populacional Brasileiro:

* Calcule a taxa de crescimento anual para o Brasil inteiro e para cada estado.
* Identifique as cidades com maior e menor crescimento em cada período censitário.
* Visualize o crescimento populacional em um gráfico de linhas ou barras.

In [4]:
pop_1991 = df[df['year'] == 1991]['population'].sum()
pop_2000 = df[df['year'] == 2000]['population'].sum()
pop_2010 = df[df['year'] == 2010]['population'].sum()
pop_2022 = df[df['year'] == 2022]['population'].sum()

## Taxa de Crescimento Anual (TCA)

A fórmula geral para calcular a TCA é:

\begin{equation}
TCA = \left( \frac{P_{\text{final}}}{P_{\text{inicial}}} \right)^{\frac{1}{n}} - 1
\end{equation}

Onde:
- P final: população no final do período
- P inicial: população no início do período
- n: número de anos entre o início e o final do período

Aplicando esta fórmula aos dados disponíveis, temos:

- **TCA (1991-2000)**: X.XX% ao ano (Substitua X.XX pelo valor calculado)
- **TCA (2000-2010)**: Y.YY% ao ano (Substitua Y.YY pelo valor calculado)
- **TCA (2010-2022)**: Z.ZZ% ao ano (Substitua Z.ZZ pelo valor calculado)

### Criar tabela com os dados calculados

In [5]:
data = {
    'year': [1991, 2000, 2010, 2022],
    'population': [pop_1991, pop_2000, pop_2010, pop_2022]
}

brazil_population_df = pd.DataFrame(data)
print(brazil_population_df)

   year   population
0  1991  146186030.0
1  2000  169252872.0
2  2010  190073788.0
3  2022  202066275.0


### Crescimento Populacional Nacional

Com os números dos últimos Census(desde 1991) podemos constatar que as taxas de Crescimento Populacional brasileira está caindo, 

In [13]:
def tca(p_initial, p_final, n):
    return (p_final/p_initial)**(1/n) - 1

# Calculando a TCA para os diferentes períodos
tca_1991_2000 = tca(pop_1991, pop_2000, 2000-1991)
tca_2000_2010 = tca(pop_2000, pop_2010, 2010-2000)
tca_2010_2022 = tca(pop_2010, pop_2022, 2022-2010)

tca_1991_2022 = tca(pop_1991, pop_2022, 2022-1991)

# Mostrando as equações
print(f"TCA (1991-2000) = {tca_1991_2000*100:.2f}% ao ano")
print(f"TCA (2000-2010) = {tca_2000_2010*100:.2f}% ao ano")
print(f"TCA (2010-2022) = {tca_2010_2022*100:.2f}% ao ano")

print(f"TCA (1991-2022) = {tca_1991_2022*100:.2f}% ao ano")

TCA (1991-2000) = 1.64% ao ano
TCA (2000-2010) = 1.17% ao ano
TCA (2010-2022) = 0.51% ao ano
TCA (1991-2022) = 1.05% ao ano


#### O que dizem os dados?

* TCA (1991-2000) = 1.64% a.a.
* TCA (2000-2010) = 1.17% a.a.
* TCA (2010-2022) = 0.51% a.a.
* TCA (1991-2022) = 1.05% a.a.

Os dados mostram que a Taxa de Crescimento Populacional está desacelerando, entre os Censo 1991 e 2000, a taxa de crescimento foi 1.64% a.a., enquanto de 2000-2010 era de 1.17% a.a. e já entre os ultimos dois censo (2010-2022) está em 0.51% a.a. sendo menor até que o período médio de 1991 e 2022 de 1.05% a.a.

O que nos leva a inferir que a população brasileira está se tornando mais velha.